matron/src/weaver.c

```diff
@@ -1032,14 +1033,34 @@ int _screen_close(lua_State *l) {
 int _screen_text_extents(lua_State *l) {
     lua_check_num_args(1);
     const char *s = luaL_checkstring(l, 1);
+    
     screen_event_text_extents(s);
-    lua_settop(l, 0);
-    return 0;
+    fprintf(stderr, "text_extents, waiting on results\n");
+    screen_results_wait();
+    fprintf(stderr, "text_extents, got screen results\n");
+    union event_data *ev = screen_results_get();
+    struct event_screen_result_text_extents *ext = (struct event_screen_result_text_extents *)ev;
+#if 1 // legacy API
```

Comment:
Could we use an additional argument to the lua function to distinguish the API call? Or is supporting code for the async API ripped out completely for now?

Reply:
ripped it out. the async version seems like more trouble than its worth. if someone is set on optimization, best would be to call the blocking function and cache results. (it _is_ silly to call it for every update with the same string value and font selection.)

that said i dpoubt this one blocking call really takes very long. maybe should time it.